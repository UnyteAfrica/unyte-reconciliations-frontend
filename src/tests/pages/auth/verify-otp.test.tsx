import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import userEvent from "@testing-library/user-event";
import React from "react";
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import { VerifyOTPPage } from "@/pages/auth/verify-otp.page";

function setup(reactNode: React.ReactNode) {
  const queryClient = new QueryClient();
  return {
    user: userEvent.setup(),
    ...render(
      <QueryClientProvider client={queryClient}>
        {reactNode}
      </QueryClientProvider>
    ),
  };
}

describe("Company Verify OTP Page", () => {
  it("renders correctly", () => {
    const tree = setup(
      <BrowserRouter>
        <VerifyOTPPage />
      </BrowserRouter>
    );
    expect(tree).toMatchSnapshot();
  });

  it("should validate OTP", async () => {
    const { user } = setup(
      <BrowserRouter>
        <VerifyOTPPage />
      </BrowserRouter>
    );

    const otpInput = screen.getByLabelText("OTP");
    await user.click(otpInput);
    await user.keyboard("2244");
    const submitButton = screen.getByText("Verify");
    await user.click(submitButton);
    const errorMessage = screen.getByText(
      "OTP cannot be less than 6 characters"
    );
    expect(errorMessage).toBeVisible();
  });
});
