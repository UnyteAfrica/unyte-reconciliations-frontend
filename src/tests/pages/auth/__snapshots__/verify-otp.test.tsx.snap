// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Company Verify OTP Page > renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="px-5 py-10 max-w-[600px] mx-auto min-h-svh flex flex-col"
      >
        <img
          alt=""
          class="mb-6 block w-28"
          src="/src/assets/Icons/UnyteLogo.svg"
        />
        <div
          class="grow"
        />
        <form>
          <header>
            <p
              class="mb-2 font-semibold text-[20px] text-center"
            >
              Verify OTP
            </p>
          </header>
          <div
            class="space-y-6"
          >
            <div>
              <label
                class="mb-2 block text-sm text-[#333"
                for="otp"
              >
                OTP
              </label>
              <div
                class="flex relative items-center w-[100%] border rounded-lg transition duration-300 focus-within:border-black p-2 h-[58px] border-[#E0E0E0]"
              >
                <div
                  class="w-full"
                >
                  <input
                    class="block text-lg font-sora px-3 py-[16px] transition border-black/30 rounded-lg outline-none autofill:shadow-[inset_0_0_0px_1000px_rgb(255,255,255)] mb-0 border-0 w-[90%] focus:border-transparent focus:border-b-0 h-[56px]"
                    id="otp"
                    name="otp"
                    placeholder="******"
                    type="password"
                    value=""
                  />
                </div>
                <button
                  class="absolute top-[50%] -translate-y-[50%] right-3"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 576 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M288 144a110.94 110.94 0 0 0-31.24 5 55.4 55.4 0 0 1 7.24 27 56 56 0 0 1-56 56 55.4 55.4 0 0 1-27-7.24A111.71 111.71 0 1 0 288 144zm284.52 97.4C518.29 135.59 410.93 64 288 64S57.68 135.64 3.48 241.41a32.35 32.35 0 0 0 0 29.19C57.71 376.41 165.07 448 288 448s230.32-71.64 284.52-177.41a32.35 32.35 0 0 0 0-29.19zM288 400c-98.65 0-189.09-55-237.93-144C98.91 167 189.34 112 288 112s189.09 55 237.93 144C477.1 345 386.66 400 288 400z"
                    />
                  </svg>
                </button>
              </div>
              <p
                class="font-sora text-sm text-[#333]"
              />
            </div>
            <div>
              <button
                class="text-mPrimary mb-2"
              >
                Resend OTP
              </button>
              <button
                class="w-full font-medium text-xl leading-[24px] bg-mPrimary h-[58px] text-white rounded-2xl"
              >
                Verify
              </button>
            </div>
          </div>
        </form>
        <div
          class="grow"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="px-5 py-10 max-w-[600px] mx-auto min-h-svh flex flex-col"
    >
      <img
        alt=""
        class="mb-6 block w-28"
        src="/src/assets/Icons/UnyteLogo.svg"
      />
      <div
        class="grow"
      />
      <form>
        <header>
          <p
            class="mb-2 font-semibold text-[20px] text-center"
          >
            Verify OTP
          </p>
        </header>
        <div
          class="space-y-6"
        >
          <div>
            <label
              class="mb-2 block text-sm text-[#333"
              for="otp"
            >
              OTP
            </label>
            <div
              class="flex relative items-center w-[100%] border rounded-lg transition duration-300 focus-within:border-black p-2 h-[58px] border-[#E0E0E0]"
            >
              <div
                class="w-full"
              >
                <input
                  class="block text-lg font-sora px-3 py-[16px] transition border-black/30 rounded-lg outline-none autofill:shadow-[inset_0_0_0px_1000px_rgb(255,255,255)] mb-0 border-0 w-[90%] focus:border-transparent focus:border-b-0 h-[56px]"
                  id="otp"
                  name="otp"
                  placeholder="******"
                  type="password"
                  value=""
                />
              </div>
              <button
                class="absolute top-[50%] -translate-y-[50%] right-3"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 576 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M288 144a110.94 110.94 0 0 0-31.24 5 55.4 55.4 0 0 1 7.24 27 56 56 0 0 1-56 56 55.4 55.4 0 0 1-27-7.24A111.71 111.71 0 1 0 288 144zm284.52 97.4C518.29 135.59 410.93 64 288 64S57.68 135.64 3.48 241.41a32.35 32.35 0 0 0 0 29.19C57.71 376.41 165.07 448 288 448s230.32-71.64 284.52-177.41a32.35 32.35 0 0 0 0-29.19zM288 400c-98.65 0-189.09-55-237.93-144C98.91 167 189.34 112 288 112s189.09 55 237.93 144C477.1 345 386.66 400 288 400z"
                  />
                </svg>
              </button>
            </div>
            <p
              class="font-sora text-sm text-[#333]"
            />
          </div>
          <div>
            <button
              class="text-mPrimary mb-2"
            >
              Resend OTP
            </button>
            <button
              class="w-full font-medium text-xl leading-[24px] bg-mPrimary h-[58px] text-white rounded-2xl"
            >
              Verify
            </button>
          </div>
        </div>
      </form>
      <div
        class="grow"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
  "user": {
    "clear": [Function],
    "click": [Function],
    "copy": [Function],
    "cut": [Function],
    "dblClick": [Function],
    "deselectOptions": [Function],
    "hover": [Function],
    "keyboard": [Function],
    "paste": [Function],
    "pointer": [Function],
    "selectOptions": [Function],
    "setup": [Function],
    "tab": [Function],
    "tripleClick": [Function],
    "type": [Function],
    "unhover": [Function],
    "upload": [Function],
  },
}
`;
